File: 241215/정수 명령 처리 2/process-numeric-commands-2.cpp
#include <iostream>
#include <queue>
#include <string>
using namespace std;

int n,num;
queue<int> qu;
int main() {
    cin >> n;
    for(int i=0; i<n; i++){
        string s;
        cin >> s;
        if(s == "push"){
            cin >> num;
            qu.push(num);
        } 
        else if(s == "front") cout << qu.front() <<'\n';
        else if(s == "size") cout << qu.size() <<'\n';
        else if(s == "pop"){
            cout << qu.front()<<'\n';
            qu.pop();
        } 
        else if(s == "empty") cout << qu.empty() <<'\n';
    }
    return 0;
}

File: 250221/정수 명령 처리 2/process-numeric-commands-2.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Queue<Integer> q = new LinkedList<>();
        int n = sc.nextInt();

        for(int i=0; i<n; i++){
            String s = sc.next();
            if(s.equals("push")){
                int num = sc.nextInt();
                q.add(num);
            }
            else if(s.equals("front")) System.out.println(q.peek());
            else if(s.equals("size")) System.out.println(q.size());
            else if(s.equals("empty")) {
                if(q.isEmpty()) System.out.println(1);
                else System.out.println(0);
            }
            else System.out.println(q.poll());
        }
        sc.close();
    }
}